### PR TITLE
Deletable ephemeral messages

### PR DESCRIPTION
**Pull request by dookon**
see https://github.com/discord/discord-api-docs/commit/8ebc79d47d554d0ba557e2a9103129ee83372fbc

### CODE REVIEW

File: core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt [hunk 1]
```diff
@@ -22,6 +22,14 @@ import kotlin.contracts.contract
  */
 public interface MessageInteractionResponseBehavior : FollowupPermittingInteractionResponseBehavior {
 
+    /**
+     * Requests to delete the message.
+     * @throws RestRequestException if something went wrong during the request.
+     */
```
Comment (Lukellmann) [inline core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt:28]:
```suggestion
    /**
     * Requests to delete the message.
     *
     * This [MessageInteractionResponseBehavior] can still be used to send followup messages to the interaction.
     *
     * @throws RestRequestException if something went wrong during the request.
     */
```
Or was there a reason that you left this part out?

**Review by Lukellmann — CHANGES_REQUESTED** [review]
Looks good overall, just two little things, see comments.
Have you also tested that it works?

File: core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt (hunk 1 above)
Comment (dookon) [reply core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt:28]: I didn't understand this "This [MessageInteractionResponseBehavior] can still..." part being documented in this function, so I removed it, but I can put it back in

File: core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt [hunk 2]
```diff
@@ -22,6 +22,14 @@ import kotlin.contracts.contract
...
+    public suspend fun delete() {
+        kord.rest.interaction.deleteOriginalInteractionResponse(applicationId, token)
+    }
```
[the remark this answers is not in the document]
Comment (dookon) [reply core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt:31]: Sure!

**Comment by dookon** [comment]
> Looks good overall, just two little things, see comments. Have you also tested that it works?

I tested it, it works

File: core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt (hunk 1 above)
Comment (Lukellmann) [reply core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt]: It means that after you run code like this:
```kotlin
val response: MessageInteractionResponseBehavior = interaction.respondPublic { /* ... */ }
response.delete()
```
you can still do this:
```kotlin
response.createPublicFollowup { /* ... */ }
```

If you can think of a way to better describe this, feel free to add it.

**Review by Lukellmann — APPROVED** [review]
lgtm and should be ready to merge except you want to clarify documentation for [this](https://github.com/kordlib/kord/pull/706#discussion_r1006360317).
Thanks!

File: core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt (hunk 1 above)
Comment (dookon) [reply core/src/main/kotlin/behavior/interaction/response/MessageInteractionResponseBehavior.kt]: aaah, makes sense

**Comment by dookon** [comment]
Able to merge.

**Pull request merged**